nipype/caching/memory.py

```diff
@@ -293,7 +294,7 @@ def _clear_all_but(self, runs, warn=True):
             input.
         """
         rm_all_but(self.base_dir, set(runs.keys()), warn=warn)
-        for dir_name, job_names in runs.items():
+        for dir_name, job_names in list(runs.items()):
```

Comment:
I think this is one of the safe operations of running ```futurize --stage1```. So, answering you question, no, this did not come up from a test.